src/main/webapp/index.jsp

```diff
@@ -1,141 +1,100 @@
-        right: 5px;
-        top: 60%;
-        margin-top: -195px;
-        height: 330px;
-        width: 200px;
-        padding: 5px;
-        z-index: 5;
-        border: 1px solid #999;
-        background: #fff;
-      }
-      h2 {
-        font-size: 22px;
-        margin: 0 0 5px 0;
-      }
-      ul {
-        list-style-type: none;
-        padding: 0;
-        margin: 0;
-        height: 271px;
-        width: 200px;
-        overflow-y: scroll;
-      }
-      li {
-        background-color: #f1f1f1;
-        padding: 10px;
-        text-overflow: ellipsis;
-        white-space: nowrap;
-        overflow: hidden;
-      }
-      li:nth-child(odd) {
-        background-color: #fcfcfc;
-      }
-      #more {
-        width: 100%;
-        margin: 5px 0 0 0;
-      }
-    </style>
     <script>
-      "use strict";
-      // This example requires the Places library. Include the libraries=places
-      // parameter when you first load the API. For example:
-      // <script src="https://maps.googleapis.com/maps/api/js?key=YOUR_API_KEY&libraries=places">
+      var infoWindow, map, pos;
+
       function initMap() {
         // Create the map.
-        const newyork = {
+        var newyork = {
           lat: 40.7591703,
           lng: -74.0394425
         };
-        const map = new google.maps.Map(document.getElementById("map"), {
+
+        map = new google.maps.Map(document.getElementById("map"), {
           center: newyork,
           zoom: 17
-        }); // Create the places service.
+        });
+
+        infoWindow = new google.maps.InfoWindow;
+        doGeolocation(infoWindow, map, pos);
+
+        //Draw the circle and populate the search results
+        const service = new google.maps.places.PlacesService(map);
+        document.getElementById("results").addEventListener("click", () => {  
+            createCityCircle(pos);
+            doNearbySearch(service, map);
+        });
+
+        //change the map to the input location
+        const geocoder = new google.maps.Geocoder();
+        document.getElementById("submit").addEventListener("click", () => {
+          geocodeAddress(geocoder, map);
+        });
+      }//init map
+
+    function doGeolocation(infoWindow, map, pos){
+        if (navigator.geolocation) {
+          navigator.geolocation.getCurrentPosition(function(position) {
+            pos = {
+              lat: position.coords.latitude,
+              lng: position.coords.longitude
+            };
+
+            infoWindow.setPosition(pos);
+            infoWindow.setContent('Location found.');
+            infoWindow.open(map);
+            map.setCenter(pos);
+
+          }, function() {
+            handleLocationError(true, infoWindow, map.getCenter());
+          });
+        } else {
+          // Browser doesn't support Geolocation
+          handleLocationError(false, infoWindow, map.getCenter());
+        }//if and else
+    }//doGeolocation
+
+    function doNearbySearch(service, map){
```

Comment:
this doesn't seem to be automatically called whenever "generate results" is pressed, which means pressing the button just draws the circle without fetching results

Reply:
This should be changed now, I have everything being manually called until we get a working UI and then we can start rerouting things based on how we want the display to look